src/ray/core_worker/reference_count.cc

```diff
@@ -899,31 +913,31 @@ void ReferenceCounter::SetReleaseLineageCallback(
 void ReferenceCounter::AddObjectLocation(const ObjectID &object_id,
                                          const NodeID &node_id) {
   absl::MutexLock lock(&mutex_);
-  auto it = object_id_locations_.find(object_id);
-  if (it == object_id_locations_.end()) {
-    it = object_id_locations_.emplace(object_id, absl::flat_hash_set<NodeID>()).first;
-  }
-  it->second.insert(node_id);
+  auto it = object_id_refs_.find(object_id);
+  RAY_CHECK(it != object_id_refs_.end())
```

Comment:
I don't think we want to make this an assertion because I believe this can actually happen if the ref goes out of scope before we get the RPC from plasma.

Reply:
Perhaps we could instead return an error code in the RPC response if the ref is out of scope?

Reply:
That's a great point!